File: Yow.CoD.Finance.Domain/Contracts/CreateLoanCommand.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public class CreateLoanCommand : Command
    {
        public CreateLoanCommand(Guid commandId, 
            Guid aggregateId,
            DateTimeOffset createdOn,
            CustomerContact customerContact, 
            BankAccount bankAccount, PaymentPlan paymentPlan, decimal amount, Duration term) 
            : base(commandId, aggregateId)
        {
            CustomerContact = customerContact;
            BankAccount = bankAccount;
            PaymentPlan = paymentPlan;
            Amount = amount;
            Term = term;
            CreatedOn = createdOn;
        }

        public DateTimeOffset CreatedOn { get; }

        public CustomerContact CustomerContact { get; }

        public BankAccount BankAccount { get; }

        public PaymentPlan PaymentPlan { get; }

        public decimal Amount { get; }

        public Duration Term { get; }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/contracts/CreateLoanCommand.java
package com.leecampbell.cod.domain.contracts;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public final class CreateLoanCommand extends DomainCommand
{
    
    private final OffsetDateTime createdOn;
    private final CustomerContact customerContact;
    private final BigDecimal amount;
    
    //TODO: Complete this port -LC
    public CreateLoanCommand(
        UUID commandId, 
        UUID aggregateId,
        OffsetDateTime createdOn, 
        CustomerContact customerContact,
        BankAccount bankAccount,
        PaymentPlan paymentPlan,
        BigDecimal amount,
        Duration term) {
            super(commandId, aggregateId);

            if (createdOn == null) throw new IllegalArgumentException("createdOn cannot be null");
            if (customerContact == null) throw new IllegalArgumentException("customerContact cannot be null");
            if (bankAccount == null) throw new IllegalArgumentException("bankAccount cannot be null");
            if (paymentPlan == PaymentPlan.None) throw new IllegalArgumentException("paymentPlan cannot be None");
            if (amount == null) throw new IllegalArgumentException("amount cannot be null");
            if (term == null) throw new IllegalArgumentException("term cannot be null");


            this.createdOn = createdOn;
            this.customerContact = customerContact;
            this.amount = amount;
    }

    public OffsetDateTime createdOn(){
        return createdOn;
    }

    public CustomerContact customerContact(){
        return customerContact;
    }
    
    public BigDecimal amount(){
        return amount;
    }
}
